src/simple.ts

```diff
@@ -3,40 +3,48 @@ import { BytesLike, HexString, toHex } from './bytes';
 import { MultiProof, processProof, processMultiProof } from './core';
 import { MerkleTreeData, MerkleTreeImpl } from './merkletree';
 import { MerkleTreeOptions } from './options';
+import { NodeHash } from './hashes';
 import { validateArgument } from './utils/errors';
 
 export interface SimpleMerkleTreeData extends MerkleTreeData<HexString> {
   format: 'simple-v1';
+  hash?: 'custom';
 }
 
 export function formatLeaf(value: BytesLike): HexString {
   return defaultAbiCoder.encode(['bytes32'], [value]);
 }
 
 export class SimpleMerkleTree extends MerkleTreeImpl<BytesLike> {
-  static of(values: BytesLike[], options: MerkleTreeOptions = {}): SimpleMerkleTree {
-    const [tree, indexedValues] = MerkleTreeImpl.prepare(values, options, formatLeaf);
-    return new SimpleMerkleTree(tree, indexedValues, formatLeaf);
+  static of(values: BytesLike[], options: MerkleTreeOptions & { nodeHash?: NodeHash } = {}): SimpleMerkleTree {
```

Comment:
We can make this a type:

```typescript
interface SimpleMerkleTreeOptions extends MerkleTreeOptions {
  nodeHash?: NodeHash
}
```

Then:

```suggestion
  static of(values: BytesLike[], options: SimpleMerkleTreeOptions = {}): SimpleMerkleTree {
```